engine/src/main/scala/cromwell/services/EngineMetadataServiceActor.scala

```diff
@@ -88,9 +95,14 @@ case class EngineMetadataServiceActor(serviceConfig: Config, globalConfig: Confi
     case GetStatus(workflowId) => queryStatusAndRespond(workflowId)
     case WorkflowQuery(uri, parameters) => queryWorkflowsAndRespond(uri, parameters)
     case RefreshSummary => summaryActor ! SummarizeMetadata
-    case MetadataSummarySuccess => scheduleSummary
+    case MetadataSummarySuccess =>
+      workflowExistenceCache = workflowExistenceCache mapValues { _ - 1 } filter { case (_, v) => v > 0}
```

Comment:
I'm not 100% sure how the whole `mapValues`/view thing works (except that one shouldn't put non-RT functions in there, which you're not) but I *think* this could be a giant performance issue - if it works the way I think it does, every one of these calls will store the latest changes which will be reapplied on every read.

disclaimer: there's a 82.81222% change it doesn't work the way I think it does